### ISSUE TITLE
Remove required attribute when chosen is enabled

### ISSUE THREAD

**Issue by ghost** [contao-components/chosen#2]
<a href="https://github.com/Toflar"><img src="https://avatars.githubusercontent.com/u/481937?v=3" align="left" width="42" height="42"></img></a> [Issue](https://github.com/contao/core-bundle/issues/423) by @Toflar
November 25th, 2015, 09:40 GMT

This will prevent the `unfocusable` javascript error in Chrome. We'll have to see if we can switch to a different script than Chosen in the future which supports html5 validation as well (and as a best case scenario is vanilla JS).

Not exactly sure how to fix this though. What do you think about this:

``` diff
diff --git a/src/Resources/contao/widgets/SelectMenu.php b/src/Resources/contao/widgets/SelectMenu.php
index 1428147..4f39f8b 100644
--- a/src/Resources/contao/widgets/SelectMenu.php
+++ b/src/Resources/contao/widgets/SelectMenu.php
@@ -152,6 +152,7 @@ class SelectMenu extends \Widget
        if ($this->chosen)
        {
            $strClass .= ' tl_chosen';
+           unset($this->arrAttributes['required']);
        }

        return sprintf('<select name="%s" id="ctrl_%s" class="%s%s"%s onfocus="Backend.getScrollOffset()">%s</select>%s',
```

**Comment by leofeyer** [comment]
As discussed, we want to fix this on JS level:

``` js
.setStyle('display', 'none').removeProperty('required')
```

**Comment by leofeyer** [comment]
Fixed in contao/core@932ecf6364cac5a1ffd14310013288060c435437 and b54cd06a6ea8bf9b8d36be72472f0f4d378de246.
